godtools/App/Features/AppLanguage/Presentation/DownloadableLanguages/Subviews/LanguageDownloadIcon.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  LanguageDownloadIcon.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 12/7/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import SwiftUI
+
+struct LanguageDownloadIcon: View {
+    
+    private static let lightGrey = Color.getColorWithRGB(red: 151, green: 151, blue: 151, opacity: 1)
+    
+    private let languageDownloadStatus: LanguageDownloadStatusDomainModel
+    
+    init(languageDownloadStatus: LanguageDownloadStatusDomainModel) {
+        self.languageDownloadStatus = languageDownloadStatus
+    }
+
+    var body: some View {
+        
+        ZStack {
+            
+            Circle()
+                .stroke(lineWidth: 1.3)
+            
+            innerIcon()
+        }
+        .frame(width: 21, height: 21)
+        .foregroundColor(iconColor())
+    }
+    
+    @ViewBuilder private func innerIcon() -> some View {
+        
+        switch languageDownloadStatus {
+        case .notDownloaded:
+            
+            Image(systemName: "arrow.down.to.line")
+                .imageScale(.small)
+            
+        case .downloading(let progress):
+            
+            drawProgressInCircle(progress: progress)
+            
+        case .downloaded:
+            
+            Image(systemName: "checkmark")
+                .imageScale(.small)
+        }
+    }
+    
+    @ViewBuilder private func drawProgressInCircle(progress: Double) -> some View {
```

Comment:
Nice work on this, I like the circular animation effect.  We may at some point want to pull some of this out into a more generic shared view if we have a need to re-use this, but for now this is great.